src/main/java/lotto/model/WinningResult.java

```diff
@@ -0,0 +1,51 @@
+package lotto.model;
+
+import java.util.EnumMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+public class WinningResult {
+
+    private final Map<Rank, Long> result;
+
+    public WinningResult(WinningNumbers winningNumbers, List<LottoNumbers> lottoNumbers) {
+        List<MatchCount> matchCounts = calculateWinningCount(winningNumbers, lottoNumbers);
+        this.result = calculateResults(matchCounts);
+    }
+
+    public Map<Rank, Long> getWinningStatistics() {
+        return new EnumMap<>(result);
+    }
+
+    public Prize calculateTotalPrize() {
+        return result.entrySet().stream()
+                .map(entry -> entry.getKey().getPrize().multiply(entry.getValue()))
+                .reduce(new Prize(0), (prize1, prize2) ->
+                    new Prize(new Money(prize1.getAmount().getAmount() +
+                                      prize2.getAmount().getAmount())));
```

Comment:
다음과 같이 구현할 수 있을 것 같네요!

```suggestion
        return result.entrySet().stream()
                .map(entry -> entry.getKey().getPrize().multiply(entry.getValue()))
                .map(Prize::getAmount)
                .reduce(Money::sum)
                .map(Prize::new)
                .orElseGet(() -> new Prize(0));
```
```java
public class Money {
    public Money sum(Money other) {
        return new Money(amount + other.amount);
    }
}
```
또한 `Prize`와 `Money`를 구분하지 않았다면, 더 간결하게 구현할 수 있었을지 모르겠군요. 😂